feat: add cilogon login and azcopy features

kirenotneb opened the pull request.

megasanjay commented on prisma/schema.prisma:
```diff
@@ -9,12 +9,6 @@ datasource db {
   schemas  = ["download", "public"]
 }
 
-model alembic_version {
-  version_num String @id(map: "alembic_version_pkc") @db.VarChar(32)
-
-  @@schema("public")
-}
-
```
This will need to exist since we use alembic for API migrations

kirenotneb replied:
This model should be back now

megasanjay reviewed (changes requested):
Requesting a minor change to prisma schema

heathharrelson reviewed (approved):
This looks how I would expect, but let's make sure we address Sanjay's comment. Since these changes are coming from staging, the schema will need to be fixed there as well.

kirenotneb commented:
I agree with the prisma changes. We seem to have a lot of churn in this area that continues to burn us. I'll take a look and see if @ejdysinger and I can get this straightened out before a merge. I already addressed the duplicate enum (not sure how that happened), but I want to cast another eye on the schema.prisma again as this PR introcued face more changes there that I expected.

megasanjay commented:
@kirenotneb I don't think you need to add an alembic model for the download schema. We only use the public schema in our API with alembic so the download items shouldn't be touched

kirenotneb commented:
> @kirenotneb I don't think you need to add an alembic model for the download schema. We only use the public schema in our API with alembic so the download items shouldn't be touched

Gotcha - it was added when I used `prisma db pull` so I thought it would be useful to included it for consistency's sake. I don't think it makes a difference for anything on the download portion so I can remove it again.

megasanjay commented:
oh i wonder if that exists on the download schema already? I might need to doublecheck this

kirenotneb commented:
> oh i wonder if that exists on the download schema already? I might need to doublecheck this

@megasanjay - it does exist already since we are using alembic with the pipeline now to manage the database structure for that schema. That said it doesn't impact anything by not having it in prisma as we added it to the download schema a number of weeks ago.

If you don't have any other concerns we can look at adding it back in the future since I just removed it. Once this PR is approved I can update the watermark pipeline to add support for azcopy and cilogon and then we can push all of this to production.

megasanjay commented:
sounds good.
